src/crimson/osd/osd.cc

```diff
@@ -103,6 +103,8 @@ OSD::OSD(int id, uint32_t nonce,
     tick_timer{[this] {
       std::ignore = update_heartbeat_peers(
       ).then([this] {
+        return deep_scrub_purged_snaps();
```

Comment:
This should probably be conditional on a config option.  Does classic actually do this unconditionally?

Reply:
yes, but it is not unconditional, it will compute the time interval in the method `deep_scrub_purged_snaps`.